core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -288,19 +288,23 @@ public void getStatus() throws Exception {
     return conf;
   }
 
-  private void mockFileSystemContextAndMasterClient() {
+  private void mockFileSystemContextAndMasterClient() throws Exception {
     mMockFileSystemContext = Mockito.mock(FileSystemContext.class);
-    AbstractFileSystem.setFileSystemContext(mMockFileSystemContext);
+    PowerMockito.mockStatic(FileSystemContext.class);
+    PowerMockito.when(FileSystemContext.create(Mockito.any(Subject.class)))
+        .thenReturn(mMockFileSystemContext);
     mMockFileSystemMasterClient = Mockito.mock(FileSystemMasterClient.class);
     Mockito.when(mMockFileSystemContext.acquireMasterClient())
         .thenReturn(mMockFileSystemMasterClient);
+    Mockito.doNothing().when(mMockFileSystemMasterClient).connect();
   }
 
-  private void mockUserGroupInformation() throws IOException {
+  private void mockUserGroupInformation(String username) throws IOException {
```

Comment:
not sure whether it's easy to create a unit test to simulate a UGI with proxy user, which might be helpful to verify the comments regarding `getRealUser` above